phd-tests/framework/src/serial/vt80x24.rs

```diff
@@ -148,3 +152,23 @@ fn make_absolute_cursor_position(col: usize, row: usize) -> Change {
         y: Position::Absolute(row),
     }
 }
+
+fn trace_buffer_contents(contents: &str) {
+    let mut last_non_empty = None;
+    for (idx, line) in contents.lines().enumerate() {
+        if line.chars().any(|c| c != ' ') {
+            last_non_empty = Some(idx);
+        }
+    }
```

Comment:
style nit, take it or leave it: this looks like [`Iterator::position`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.position):
```suggestion
    let last_non_empty =  contents.lines().position(|line| line.chars().any(|c| c != ' '));
```